### PR TITLE
GPHDFS to use buffered i/o when reading Parquet

### PR DESCRIPTION
**Pull request by sansanichfb**
The way currently Parquet reader works in GPHDFS - it uses stdout as a way to transfer data from GPHDFS JVM back to GPDB. Given the fact that System.out is unbuffered, and GPHDFS writes ot System.out each tuple - making System.out buffered gives noticable performance gain.

PR contains code change and basic regression tests, which execrices reading integer data type from Parquet and logic for installing Parquet jar into Hadoop environment.

### CODE REVIEW

File: gpAux/extensions/gphdfs/src/java/1.2/com/emc/greenplum/gpdb/hdfsconnector/HDFSReader.java [hunk 1]
```diff
@@ -163,7 +163,7 @@ else if (isAvroFormat) {
 			avroReader.readAvroFormat();
 		}
 		else if (isParquetFormat) {
-			GpdbParquetFileReader parquetReader = new GpdbParquetFileReader(conf, segid, totalseg, inputpath, tableSchema, schemaComplete, autoSelect, System.out);
+			GpdbParquetFileReader parquetReader = new GpdbParquetFileReader(conf, segid, totalseg, inputpath, tableSchema, schemaComplete, autoSelect, new BufferedOutputStream(System.out));
```
Comment (lavjain) [inline gpAux/extensions/gphdfs/src/java/1.2/com/emc/greenplum/gpdb/hdfsconnector/HDFSReader.java]: BufferedOutputStream can be deferred to GpdbParquetFileReader class <line 106>

DataOutputStream dos = new DataOutputStream(new BufferedOutputStream(out));

**Comment by lavjain** [comment]
LGTM

File: gpAux/extensions/gphdfs/regression/output/int_ExtreadwriteParquet.source [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ count
+-------
+  5000
+(1 row)
+
+--start_ignore
+\!/tmp/hadoop-2.7.3/bin/hadoop fs -rm -r /mapreduce/*
+rm: `/mapreduce/*': No such file or directory
+\!/tmp/hadoop-2.7.3/bin/hadoop fs -rm -r /mapred/*
+rm: `/mapred/*': No such file or directory
```
Comment (denalex) [inline gpAux/extensions/gphdfs/regression/output/int_ExtreadwriteParquet.source]: just wonder who issues these removes and what are those ignore comments for ?

File: gpAux/extensions/gphdfs/src/java/1.2/com/emc/greenplum/gpdb/hdfsconnector/HDFSReader.java (hunk 1 above)
Comment (denalex) [inline gpAux/extensions/gphdfs/src/java/1.2/com/emc/greenplum/gpdb/hdfsconnector/HDFSReader.java]: Would the same buffering help Avro and GPDBWriteable as well ? In other words, should we apply this logic to other places here ?

Comment (sansanichfb) [inline gpAux/extensions/gphdfs/src/java/1.2/com/emc/greenplum/gpdb/hdfsconnector/HDFSReader.java]: Could be applied to other places, but that will require additional testing, and this particular PR addresses and tests Parquet format slowness.

Comment (sansanichfb) [inline gpAux/extensions/gphdfs/src/java/1.2/com/emc/greenplum/gpdb/hdfsconnector/HDFSReader.java]: Thanks, looks cleaner, updated.

File: gpAux/extensions/gphdfs/regression/output/int_ExtreadwriteParquet.source (hunk 2 above)
Comment (sansanichfb) [inline gpAux/extensions/gphdfs/regression/output/int_ExtreadwriteParquet.source]: run_gphdfs_regression issues those statements, they meant to be ignored on a comparison.

**Pull request closed without merging**